src/charm.py

```diff
@@ -257,16 +262,36 @@ def remote_mongodb_config(self, hosts, replset=None, standalone=None) -> MongoCo
             OperatorUser, hosts, replset=replset, standalone=standalone
         )
 
+    @property
+    def _is_storage_from_different_cluster(self) -> bool:
+        """Checks if we are reusing storage from a different cluster."""
+        lock_hash = self.lock_hash  # Avoid duplicate read of the file
+        return lock_hash != UNDEFINED and lock_hash != self.databag_lock_hash
+
     @property
     def mongo_config(self) -> MongoConfiguration:
         """Returns a MongoConfiguration object for shared libs with agnostic mongo commands."""
         return self.mongodb_config
 
+    @property
+    def standalone_config(self) -> MongoConfiguration:
+        """Generates a MongoConfiguration object for local authenticated standalone connection."""
+        return self._get_mongodb_config_for_user(
+            OperatorUser, {self.unit_host(self.unit)}, standalone=True
+        )
+
     @property
     def mongodb_config(self) -> MongoConfiguration:
         """Generates a MongoConfiguration object for this deployment of MongoDB."""
         return self._get_mongodb_config_for_user(OperatorUser, set(self.app_hosts))
 
+    @property
+    def single_host_config(self) -> MongoConfiguration:
```

Comment:
```suggestion
    def unit_host_config(self) -> MongoConfiguration:
```

not sure on the naming of this, but should be updated to reflect that it using the current unit's host

Reply:
This function has been removed :)